router/batchrouter/asyncdestinationmanager/bing-ads/bulk_uploader.go

```diff
@@ -0,0 +1,296 @@
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while unmarshalling the destination config %v", err.Error()),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	var failedJobs []int64
+	var successJobs []int64
+	var importIds []string
+	var errors []string
+	actionFiles, err := b.CreateZipFile(asyncDestStruct.FileName, destConfig.AudienceID)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while transforming the file. %v", err.Error()),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	uploadRetryableStat := stats.Default.NewTaggedStat("events_over_prescribed_limit", stats.CountType, map[string]string{
+		"module":   "batch_router",
+		"destType": b.destName,
+	})
+	for _, actionFile := range actionFiles {
+		uploadRetryableStat.Count(len(actionFile.FailedJobIDs))
+		_, err := os.Stat(actionFile.ZipFilePath)
+		if err != nil || actionFile.EventCount == 0 {
+			continue
+		}
+		urlResp, err := b.service.GetBulkUploadUrl()
+		if err != nil {
+			b.logger.Error("Error in getting bulk upload url: %w", err)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:error in getting bulk upload url: %s", actionFile.Action, err.Error()))
+			continue
+		}
+
+		if urlResp.UploadUrl == "" || urlResp.RequestId == "" {
+			b.logger.Error(`{"error" : "getting empty string in upload url or request id. Provide correct configuration credentials."}`)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:getting empty string in upload url or request id. Provide correct configuration credentials.", actionFile.Action))
+			continue
+		}
+
+		uploadTimeStat := stats.Default.NewTaggedStat("async_upload_time", stats.TimerType, map[string]string{
+			"module":   "batch_router",
+			"destType": b.destName,
+		})
+
+		startTime := time.Now()
+		uploadBulkFileResp, errorDuringUpload := b.service.UploadBulkFile(urlResp.UploadUrl, actionFile.ZipFilePath)
+		uploadTimeStat.Since(startTime)
+
+		if errorDuringUpload != nil {
+			b.logger.Error("error in uploading the bulk file: %v", errorDuringUpload)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:error in uploading the bulk file: %v", actionFile.Action, errorDuringUpload))
+			continue
+		}
+
+		if uploadBulkFileResp.RequestId == "" || uploadBulkFileResp.TrackingId == "" {
```

Comment:
Should this be checked in the application or do we let bing-ads-go-sdk handle these cases too post unmarshal

Reply:
@koladilip

Reply:
we should handle in  bing-ads-go-sdk too. @ItsSudip can you have look into it?